elements/lisk-validator/src/keywords/data_type.ts

```diff
@@ -75,13 +80,40 @@ const compile = (
 		_parentDataProperty?: string | number,
 		_rootData?: object,
 	): boolean => {
-		if (value === 'boolean') return isBoolean(data);
-		if (value === 'bytes') return isBytes(data as Buffer);
-		if (value === 'string') return isString(data);
-		if (value === 'uint32') return isUInt32(data);
-		if (value === 'uint64') return isUInt64(data);
-		if (value === 'sint32') return isSInt32(data);
-		if (value === 'sint64') return isSInt64(data);
+		if (value === 'boolean') {
+			return isBoolean(data);
+		}
+		if (value === 'bytes') {
+			if (!isBytes(data as Buffer)) {
+				return false;
+			}
+			const parent = parentSchema as KVPair;
+			if (typeof parent.minLength === 'number') {
+				if ((data as Buffer).length < parent.minLength) {
+					return false;
+				}
+			}
+			if (typeof parent.maxLength === 'number') {
+				if ((data as Buffer).length > parent.maxLength) {
+					return false;
+				}
+			}
+		}
+		if (value === 'string') {
+			return isString(data);
+		}
+		if (value === 'uint32') {
+			return isUInt32(data);
+		}
+		if (value === 'uint64') {
+			return isUInt64(data);
+		}
+		if (value === 'sint32') {
+			return isSInt32(data);
+		}
+		if (value === 'sint64') {
+			return isSInt64(data);
+		}
```

Comment:
This redundant code is one of the reasons I was not convinced of introducing a custom keyword instead of using standard `type` with custom formats.